File: SourceCppProject/sources/Models/LinesStorageSet.cpp
#include "LinesStorageSet.hpp"


LinesStorageSet::LinesStorageSet()
{
}

LinesStorageSet::LinesStorageSet(const std::set<std::string>& initial_set)
{
    m_Name = std::string{"Lines Storage Set"};
    m_LinesSet = initial_set;
}

LinesStorageSet::LinesStorageSet(const std::set<std::string>& initial_set, const std::string& name)
{
    m_Name = name;
    m_LinesSet = initial_set;
}

LinesStorageSet::LinesStorageSet(const std::string& name)
{
    m_Name = name;
}

void LinesStorageSet::putContent(const std::string& content)
{
    m_LinesSet.insert(content);
}

const std::set<std::string>& LinesStorageSet::getContent() const
{
    return m_LinesSet;
}

const std::string& LinesStorageSet::getName() const
{
    return m_Name;
}

File: TextLinesComparing.Library/LinesStorageSet.cs
//-----------------------------------------------------------------------
// <copyright file="LinesStorageSet.cs" company="Demo Projects Workshop">
//     Copyright (c) Demo Projects Workshop. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

#pragma warning disable SA1600 // ElementsMustBeDocumented

namespace TextLinesComparing.Library;

public class LinesStorageSet
{
    private readonly SortedSet<string> currentLinesSet;

    public LinesStorageSet()
    {
        this.currentLinesSet = new SortedSet<string>();
        this.Name = "Lines Storage Set";
    }

    public LinesStorageSet(IEnumerable<string> sequence, string name = "Lines Storage Set")
    {
        this.currentLinesSet = new SortedSet<string>(sequence);
        this.Name = name;
    }

    public string Name { get; set; }

    public SortedSet<string> Content
    {
        get { return this.currentLinesSet; }
    }

    public void PutContent(LineInfo target_info)
    {
        this.currentLinesSet.Add(target_info.Content.Value);
    }
}
